components/server/src/server.ts

```diff
@@ -138,6 +139,18 @@ export class Server {
         // Install passport
         await this.authenticator.init(app);
 
+        // log context
+        app.use(async (req: express.Request, res: express.Response, next: express.NextFunction) => {
+            const userId = req.user ? req.user.id : undefined;
+            await runWithContext(
```

Comment:
I'm not :100: , but IIRC in order for `async` to play nice with express error handling, we need to pass `err` into `next` in case it occurs. So wrapping this in a `try ... catch(err) {next(err)}` would do.

Reply:
thanks, good catch!